File: app/src/main/java/com/krenvpravo/expensenotes/ui/MainActivityPresenter.java
package com.krenvpravo.expensenotes.ui;

import android.support.annotation.Nullable;

import com.krenvpravo.expensenotes.models.Transaction;

import java.util.List;

/**
 * @author Dmitry Borodin on 2017-01-19.
 */

public class MainActivityPresenter implements MainActivityContract.Presenter {

    private List<Transaction> transactionList;
    @Nullable
    private MainActivityContract.View view;

    @Override
    public void onViewCreated() {

    }

    @Override
    public void onViewDestroyed() {
        view = null;
    }

    @Override
    public void onAddTransactionClicked() {

    }
}

File: app/src/main/java/com/krenvpravo/expensenotes/ui/MainActivityPresenter.kt
package com.krenvpravo.expensenotes.ui

import com.krenvpravo.expensenotes.models.Transaction

/**
 * @author Dmitry Borodin on 2017-01-19.
 */

class MainActivityPresenter : MainActivityContract.Presenter {

    private val transactionList: List<Transaction>? = null
    private var view: MainActivityContract.View? = null

    override fun onViewCreated() {

    }

    override fun onViewDestroyed() {
        view = null
    }

    override fun onAddTransactionClicked() {

    }
}
